analytics-integrations/google-analytics/src/main/java/com/segment/analytics/internal/integrations/GoogleAnalyticsIntegration.java

```diff
@@ -166,6 +202,20 @@ boolean handleProductEvent(String event, String category, Properties properties)
     return false;
   }
 
+  void setCustomDimensions(HitBuilders.HitBuilder hit, Properties properties) {
```

Comment:
looks like HitBuilders.HitBuilder is protected, so I'm not sure of how to abstract this out, may just have to have some repetitive logic in `.screen` and `.track`